lib/flight_scheduler/commands/batch.rb

```diff
@@ -31,46 +31,87 @@ class Batch < Command
       NUM_REGEX = /\A\d+[km]?\Z/
 
       def run
-        job = JobsRecord.create(script: script_path, min_nodes: opts.nodes, connection: connection)
+        ensure_shebang
+        job = JobsRecord.create(script: script_path, min_nodes: min_nodes, connection: connection)
         puts "Submitted batch job #{job.id}"
       end
 
-      def script_path
-        path = args.first
-        # Handle absolute paths and explicit relative paths
-        if File.absolute_path?(path) || path[0] == '.'
-          path = File.expand_path(path)
-          File.exists?(path) ? path : raise(MissingError, "Could not locate: #{path}")
-        # Handle implicit relative paths
-        elsif File.exists?(p = File.expand_path(path))
-          p
-        # Preform a PATH lookup as a fallback
-        else
-          # Do not assume PATH is correct, only consider absolute paths to be valid
-          roots = ENV.fetch('PATH', '').split(':').select do |root|
-            File.absolute_path?(root.to_s)
-          end
-
-          # Search for the root directory path
-          root = roots.find do |r|
-            File.exists? File.join(r, path)
-          end
-
-          # Return the absolute path or error
-          root ? File.join(root, path) : raise(MissingError, "Could not locate: #{path}")
+      def ensure_shebang
+        File.open(script_path) do |file|
+          shebang = file.gets(2).to_s
+          raise InputError, <<~ERROR.chomp unless shebang == '#!'
+            The script appears to have an invalid shebang line
+          ERROR
         end
       end
 
-      def nodes_opts
-        if NUM_REGEX.match? opts.nodes
-          opts.nodes
+      def min_nodes
+        return 1 unless merged_opts.nodes
+        if NUM_REGEX.match? merged_opts.nodes
+          merged_opts.nodes
         else
           raise InputError, <<~ERROR.chomp
-            Unrecognized number syntax: #{opts.nodes}
+            Unrecognized number syntax: #{merged_opts.nodes}
             It should be a number with an optional k or m suffix.
           ERROR
         end
       end
+
+      def merged_opts
+        @merged_opts ||= begin
+          magic = CLI::MAGIC_BATCH_SLOP.parse(read_magic_arguments_string.split(' '))
+                                       .to_h
+                                       .reject { |_, v| v.nil? }
+                                       .map { |k, v| [k.to_s, v] }
+                                       .to_h
+          cli = opts.reject { |_, v| v.nil? }
+                    .map { |k, v| [k.to_s, v] }
+                    .to_h
+          Hashie::Mash.new.merge(magic).merge(cli)
+        end
+      end
+
+      def read_magic_arguments_string
+        ''.tap do |args|
+          File.open(script_path) do |file|
+            regex = /\A##{opts.comment_prefix}(?<args>.*)$/
```

Comment:
This regex looks like it would match against `#SBATCH-N 100`.  Was that intended?  I think we should force a space after the prefix, unless you have a good reason for not doing that.

Reply:
https://github.com/openflighthpc/flight-scheduler/pull/3/commits/e4477315307339232c8915f59a0acb472c18d2dd